### ISSUE TITLE
`components_within_components` causes a crash when apps use the attachment component

### ISSUE THREAD

**Issue by kevindew** [alphagov/govuk_publishing_components#1570]
In Content Publisher we've serving [500 errors](https://sentry.io/organizations/govuk/issues/1721523518/?project=1242052) when viewing the component guide. The stack trace shows this is occurring in the `components_within_component` method:

```
15:02:05 web.1    | No such file or directory @ rb_sysopen - /Users/kevindew/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/govuk_publishing_components-21.55.3/app/views/govuk_publishing_components/components/attachment/_thumbnail_document.html.erb excluded from capture: DSN not set
15:02:05 web.1    |
15:02:05 web.1    | Errno::ENOENT (No such file or directory @ rb_sysopen - /Users/kevindew/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/govuk_publishing_components-21.55.3/app/views/govuk_publishing_components/components/attachment/_thumbnail_document.html.erb):
15:02:05 web.1    |
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:135:in `read'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:135:in `components_within_component'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:116:in `block in find_partials_in'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:115:in `each'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:115:in `find_partials_in'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:100:in `find_all_partials_in'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:50:in `components_in_use_sass'
15:02:05 web.1    | govuk_publishing_components (21.55.3) app/controllers/govuk_publishing_components/component_guide_controller.rb:11:in `index'
```

This seems to be caused by the use of importing svg files which is being mistaken for being different components: https://github.com/alphagov/govuk_publishing_components/blob/aa86284e7081788b6d22c50cfb70ef3992aa7dca/app/views/govuk_publishing_components/components/_attachment.html.erb#L43-L47

Some quick ideas with @alex-ju were to only match files without an extension or a a html one (since we only consider [html.erb](https://github.com/alphagov/govuk_publishing_components/blob/aa86284e7081788b6d22c50cfb70ef3992aa7dca/app/controllers/govuk_publishing_components/component_guide_controller.rb#L133) files to be components) and checking whether [the file exists](https://github.com/alphagov/govuk_publishing_components/blob/aa86284e7081788b6d22c50cfb70ef3992aa7dca/app/controllers/govuk_publishing_components/component_guide_controller.rb#L135) before trying to read it.

**Comment by kevindew** [comment]
cc @andysellick

**Comment by alex-ju** [comment]
I take it this was fixed in #1571. Shall we close it, @andysellick/@kevindew?

**Comment by kevindew** [comment]
Oh yes this is resolved. Thanks again @andysellick.
